description: launch scripts that use removed python2 modules with python3
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/gn/src/gn/ninja_action_target_writer.cc
+++ b/tools/gn/src/gn/ninja_action_target_writer.cc
@@ -136,7 +136,13 @@ std::string NinjaActionTargetWriter::Wri
   }
 
   out_ << "  command = ";
+  std::string script_name = target_->action_values().script().GetName();
+  if (script_name == "gen_xproto.py") {
+  out_ << "python3";
+  } else {
   path_output_.WriteFile(out_, settings_->build_settings()->python_path());
+  }
+
   out_ << " ";
   path_output_.WriteFile(out_, target_->action_values().script());
   for (const auto& arg : args.list()) {
